objectherkenning_openbare_ruimte/databricks_pipelines/common/table_manager.py

```diff
@@ -0,0 +1,182 @@
+from datetime import datetime
+
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructType
+
+
+class TableManager:
+    def __init__(self, spark: SparkSession, catalog: str, schema: str):
+        self.spark = spark
+        self.catalog = catalog
+        self.schema = schema
+
+    def update_status(
+        self, table_name: str, job_process_time: datetime, exclude_ids=[]
+    ):
+        count_pending_query = f"""
+        SELECT COUNT(*) as pending_count
+        FROM {self.catalog}.{self.schema}.{table_name}
+        WHERE status = 'Pending'
+        """  # nosec
+        total_pending_before = self.spark.sql(count_pending_query).collect()[0][
+            "pending_count"
+        ]
+
+        update_query = f"""
+        UPDATE {self.catalog}.{self.schema}.{table_name}
+        SET status = 'Processed', processed_at = '{job_process_time}'
+        WHERE status = 'Pending'
+        """  # nosec
+        if exclude_ids:
+            exclude_ids_str = ", ".join(map(str, exclude_ids))
+            update_query += f" AND id NOT IN ({exclude_ids_str})"
+
+        self.spark.sql(update_query)
+
+        total_pending_after = self.spark.sql(count_pending_query).collect()[0][
+            "pending_count"
+        ]
+        updated_rows = total_pending_before - total_pending_after
+
+        print(
+            f"Updated {updated_rows} 'Pending' rows to 'Processed' in {self.catalog}.{self.schema}.{table_name}, {total_pending_after} rows remained 'Pending'."
+        )
+
+    def _load_table(self, table_name: str) -> DataFrame:
+        """
+        Loads a table from the catalog and schema.
+
+        Parameters:
+        ----------
+        table_name : str
+            The name of the table to load.
+
+        Returns:
+        -------
+        DataFrame
+            A DataFrame containing the rows from the specified table.
+        """
+        full_table_name = f"{self.catalog}.{self.schema}.{table_name}"
+        table_rows = self.spark.table(full_table_name)
+        print(f"Loaded {table_rows.count()} rows from {full_table_name}.")
+        return table_rows
+
+    def load_pending_rows_from_table(self, table_name: str) -> DataFrame:
+        """
+        Loads all rows with a 'Pending' status from the specified table in the catalog and schema.
+
+        Parameters:
+        ----------
+        table_name : str
+            The name of the table from which to load 'Pending' rows.
+
+        Returns:
+        -------
+        DataFrame
+            A DataFrame containing the rows with a 'Pending' status from the specified table.
+        """
+        table_rows = self._load_table(table_name)
+        pending_table_rows = table_rows.filter("status = 'Pending'")
+        print(
+            f"Filtered to {pending_table_rows.count()} 'Pending' rows from {self.catalog}.{self.schema}.{table_name}."
+        )
+        return pending_table_rows
+
+    def load_from_table(self, table_name: str) -> DataFrame:
```

Comment:
Is this wrapper around the actual function needed? I have the feeling the other one can become public and this be removed.

Reply:
yes it is used. tbh my main wish with this structure was to group the print lines of what and how much i loaded. 

I could delete both load_pending_from_table and load_from_table, make _load_table public and then do:
```
df = tableManager.load_table()
print(...)
df = tableManager.load_table().filter(status=Pending)
print(...)
```

then i do not need the functions anymore, but im writing these print statements 30 times and it makes the code less readable.

what do you think?